src/main/java/omero/gateway/LoginCredentials.java

```diff
@@ -63,6 +65,14 @@
     /** Whether to check the client-server versions */
     private boolean checkVersion = true;
 
+    /** Default websocket ports (this might be moved into omero.constants
+     * in future) **/
+    private static Map<String,Integer> WS_PORTS = new HashMap<>();
```

Comment:
Since the hash map is mutable, wrapping with an unmodifiable version or replacing with an enum might be worth it.

Reply:
Done. Replaced with enum.